### PR TITLE
Version 0.0.1

### PR DESCRIPTION
**Pull request by Zac-HD**
I've set this up for automatic releases; you'll just need to issue a PyPI token and set it as the `TWINE_PASSWORD` repo secret.

The only trick there is that (until https://github.com/pypa/warehouse/issues/6378) for the first release, you'll need to use a full-user-scoped token, and then once the package exists you can replace that with a single-package-scoped release-only token.  The commands in `.github/workflows/ci.yml` should also be obvious if you want to run the first one locally instead.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
```
Comment (cooperlees) [inline .github/workflows/ci.yml:13]:
```suggestion
    - uses: actions/checkout@v3
```
I'll update post merge

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v2
```
Comment (cooperlees) [inline .github/workflows/ci.yml:15]:
```suggestion
      uses: actions/setup-python@v3
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      uses: actions/setup-python@v1
+      with:
+        python-version: "3.10"
+    - name: Install tools
+      run: python -m pip install --upgrade pip setuptools wheel twine
+    - name: Upload new release
+      env:
+        TWINE_USERNAME: '__token__'
+        TWINE_PASSWORD: ${{ secrets.TWINE_PASSWORD }}
+      run: python setup.py sdist bdist_wheel && twine upload --skip-existing dist/*
```
Comment (cooperlees) [inline .github/workflows/ci.yml:62]: Should prob move this to the new "build" hotness

**Review by cooperlees — APPROVED** [review]
You sir, are a legend. Thanks for getting this started!

**Comment by Zac-HD** [comment]
🎉 

Happy for you to update everything of course, my expedient template is definitely aging.  It'd also be good to add `ignore_errors=true` to the tox `check` env, so that if `shed` makes changes we still run `flake8` in the same build.

**Pull request merged**